app/src/lib/proposals-utils.js

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react'
+import { useAragonApi } from '@aragon/api-react'
+import { calculateThreshold, getMaxConviction } from '../lib/conviction'
+import BigNumber from './bigNumber'
+
+export function getStakesAndThreshold(proposal = {}) {
+  const { appState } = useAragonApi()
+  const {
+    convictionStakes,
+    stakeToken,
+    requestToken,
+    globalParams: { alpha, maxRatio, weight },
+  } = appState
+
+  const { requestedAmount } = proposal
+
+  const stakes = convictionStakes.filter(
+    stake => stake.proposal === parseInt(proposal.id)
+  )
+  const totalTokensStaked = useMemo(() => {
+    if (!proposal) {
+      return new BigNumber('0')
+    }
+    return proposal?.stakes.reduce((accumulator, stake) => {
```

Comment:
I think we may need to add the`@babel/plugin-proposal-optional-chaining` plugin?

When running the app got the error of conditional chaining not enabled.

Reply:
not sure what is that